### ISSUE TITLE
Incompatibility with altervista.org (http_request_failed)

### ISSUE THREAD

**Issue by jamesozzie** [google/site-kit-wp#1363]
## Bug Description
Altervista.org is an Italian site that allows users to easily create a WordPress blog. The domains provided to users are based on subdomains off altervista.org (ie. mysite.altervista.org). 

Users are unable to proceed with Site Kit setup as they can't verify domain ownership, based on subdomain setup. The error encountered is below. This occurs after clicking on the "Proceed" button

```
Oops! There was a problem during set up. Please try again.
Unknown Error (code: http_request_failed).
```
Video of issue: https://recordit.co/1NvEBrKEli

WordPress support topic:
https://wordpress.org/support/topic/error-http_request_failed-during-installation/

![image](https://user-images.githubusercontent.com/41326532/78767706-d71e0680-7982-11ea-8810-f4cd2abdd983.png)

Additional information:
- Possible to reproduce, setup a site and follow the typical setup flow
- One user raised this issue in the WordPress support forums
- altervista.org redirects to it.altervista.org upon 

## Additional Context

 - PHP Version: 
 - Plugin Version [e.g. 22]

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

## QA Brief

* 

## Changelog entry

*

**Comment by aaemnnosttv** [comment]
After signing up for a site and seeing this for myself I found that the error message is this:
```
cURL error 56: Received HTTP code 403 from proxy after CONNECT
```
The request wasn't made with any proxy configuration so it seemed to be set at the environment level.

Then I saw this under **Settings > Server To Server**

![image](https://user-images.githubusercontent.com/1621608/79381641-f3b1d400-7f6a-11ea-8ba5-694d379e69c9.png)

This shows that WP can only make requests to a list of allowed domains.

It also says:
> At the moment you can establish connections only to a limited number of addresses. To prevent service abuses, in order to remove the whitelist you have to complete the identification procedure by SMS

After completing the SMS challenge it seemed that I could disable the restrictions:

![image](https://user-images.githubusercontent.com/1621608/79382147-c154a680-7f6b-11ea-86b7-812ebb8388fb.png)

However, this did not appear to work (maybe it takes time to take effect?)

I would contact Altervista.org to see if we can get `*.withgoogle.com` added to the list of allowed hosts for server to server connections.
